modules/core/src/math/matrix/vpMatrix.cpp

```diff
@@ -1548,12 +1548,7 @@ void vpMatrix::stackColumns(vpColVector &out)
   if ((out.rowNum != colNum * rowNum) || (out.colNum != 1))
     out.resize(colNum * rowNum, false, false);
 
-  double *optr = out.data;
-  for (unsigned int j = 0; j < colNum; j++) {
-    for (unsigned int i = 0; i < rowNum; i++) {
-      *(optr++) = rowPtrs[i][j];
-    }
-  }
+  memcpy(out.data, data, sizeof(double)*out.getRows());
 }
```

Comment:
I didn't test, but I think that here memcpy cannot be used. We should get column 1, then column 2... while memcpy will stack rows

Reply:
Indeed, my mistakes.